dan_layer/core/src/workers/hotstuff_waiter.rs

```diff
@@ -661,35 +696,68 @@ where
         payload_id: PayloadId,
         last_height: NodeHeight,
     ) -> Result<(), HotStuffError> {
-        // TODO maybe we should check that `from` is from someone from any committee on this transaction.
+        // If the current replica did not store the payload, we simply don't do anything else and return
+        let payload = match self.shard_store.create_read_tx()?.get_payload(&payload_id) {
+            Ok(payload) => payload,
+            Err(e) => {
+                info!(
+                    target: LOG_TARGET,
+                    "Payload = {} is missing from node shard store {}", payload_id, e
+                );
+                return Ok(());
+            },
+        };
```

Comment:
Use the `Optional` trait to change a not found error to an Option, this means that other errors are still handled correctly.

```suggestion
        // If the current replica has not stored the payload, we simply don't do anything else and return
        let payload = match self.shard_store.create_read_tx()?.get_payload(&payload_id).optional()? {
            Some(payload) => payload,
            None => {
                info!(
                    target: LOG_TARGET,
                    "Payload = {} is missing from node shard store", payload_id
                );
                return Ok(());
            },
        };
```